pkgs/data/fonts/joypixels/default.nix

```diff
@@ -1,24 +1,90 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, config
+, acceptLicense ? config.joypixels.acceptLicense or false
+}:
+
+let inherit (stdenv.hostPlatform) system;
+
+    throwSystem = throw "Unsupported system: ${system}";
+
+    systemSpecific = {
+      x86_64-darwin = rec {
+        systemTag =  "nix-darwin";
+        capitalized = systemTag;
+        ext = "ttc";
+        fontFile = "Apple%20Color%20Emoji.ttc";
+        name = "joypixels-apple-color-emoji.ttc";
+      };
+      x86_64-linux = rec {
```

Comment:
Could this be made generic? There are more and more people who use NixOS on ARM devices (e.g. `aarch64-linux`). Similarly Apple is switching to ARM too.

Reply:
How about the following?
```
{ x86_64-darwin = { ... }; }.${system} or { ... } # The NixOS version of the arguments.
```

Reply:
Maybe branching on system is clearer? I only used the attrset approach because people had me switch from branches to such attrsets when I made the wire-desktop expression darwin-compatible.

Reply:
Went ahead and implemented my suggestion.

Reply:
Not sure whether there's a way to specify the system regardless of the architecture?

Reply:
Should I just use isDarwin?

Reply:
You can switch on `stdenv.hostPlatform.parsed.kernel.name` instead. This is `linux` for any Linux and `darwin` for macOS

Reply:
👍